apps/bfd-pipeline/bfd-pipeline-rda-grpc/src/main/java/gov/cms/bfd/pipeline/rda/grpc/sink/direct/AbstractClaimRdaSink.java

```diff
@@ -298,26 +327,25 @@ List<RdaChange<TClaim>> transformMessages(String apiVersion, Collection<TMessage
    *
    * @param apiVersion appropriate string for the apiSource column of the claim table
    * @param message an RDA API message object of the correct type for this sync
-   * @return the converted claim
-   * @throws DataTransformer.TransformationException if the message is invalid
+   * @return an optional containing the converted claim if successful, {@link Optional#empty()}
+   *     otherwise
+   * @throws IOException if there was an issue writing out a {@link MessageError}
+   * @throws ProcessingException if there was an issue transforming the message
    */
   @Nonnull
   @Override
-  public RdaChange<TClaim> transformMessage(String apiVersion, TMessage message)
-      throws DataTransformer.TransformationException {
+  public Optional<RdaChange<TClaim>> transformMessage(String apiVersion, TMessage message)
+      throws IOException, ProcessingException {
     try {
       var change = transformMessageImpl(apiVersion, message);
       metrics.transformSuccesses.increment();
-      return change;
+      return Optional.of(change);
     } catch (DataTransformer.TransformationException transformationException) {
       metrics.transformFailures.increment();
-      try {
-        writeError(apiVersion, message, transformationException);
-      } catch (IOException e) {
-        transformationException.addSuppressed(e);
-      }
-      throw transformationException;
+      writeError(apiVersion, message, transformationException);
     }
+
+    return Optional.empty();
```

Comment:
I'd either move this into the `catch` or return a variable that's set within the `try`.